src/main/java/sayo/Sayo.java

```diff
@@ -0,0 +1,176 @@
+package sayo;
+
+import java.util.Scanner;
+import java.io.FileNotFoundException;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Sayo {
+
+    private static ArrayList<Task> items = new ArrayList<>();
+
+    private static String dataFilePath = "./data/sayo.txt";
+
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
+
+        loadTasksFromFile();
+
+        System.out.println("Hello! I'm Sayo and it's great to see you! \n" + "What can I do for you today? \n");
+
+        String input = " ";
+
+        do {
+            try {
+                input = scanner.nextLine().trim();
+                if (input.equals("list")) {
+                    for (int i = 0; i < items.size(); i++) {
+                        System.out.println((i + 1) + ". " + items.get(i));
+                    }
+                } else if (input.startsWith("mark")) {
+                    System.out.println("test");
+                    int index = Integer.parseInt(input.substring(5)) - 1;
+                    if (index >= 0 && index < items.size()) {
+                        items.get(index).markAsDone();
+                        System.out.println("Awesome! I've marked this task as done: ");
+                        System.out.println(items.get(index));
+                    }
+                    saveTasksToFile();
```

Comment:
Perhaps you could abstract out each of the cases here to make the method shorter.